code/portal/login.c

```diff
@@ -32,7 +32,7 @@ static struct str s_from_c_str(const char *ptr)
 static int s_cmp(struct str *str1, struct str *str2)
 {
     if (str1->len != str2->len)
-        return str1->len - str2->len;
+        return (int)str1->len - (int)str2->len;

```

Comment:
This is my bad, but it's not safe to cast to a `int`.

I guess, in this case it would be better to simply do:
```c
if (str1->len < str2->len)
    return -1;
else if (str1->len > str2->len)
    return 1;
else
    return memcmp(str1->ptr, str2->ptr, str1->len);
```

Reply:
I fixed it.